rsconnect/main.py

```diff
@@ -929,60 +929,64 @@ def deploy_quarto(
     python,
     force_generate,
     verbose,
-    directory,
+    file_or_directory,
     extra_files,
     env_vars,
     image,
 ):
     set_verbosity(verbose)
 
+    base_dir = file_or_directory
+    if not isdir(file_or_directory):
+        base_dir = dirname(file_or_directory)
+
     with cli_feedback("Checking arguments"):
-        module_file = fake_module_file_from_directory(directory)
+        module_file = fake_module_file_from_directory(file_or_directory)
```

Comment:
Using a `.qmd` file for this is fine, I'm assuming, since deployment worked correctly in my case.

Update: Looking at how `module_file` is used and what the function does., yeah, it's seems fine.